app/models/spree_stripe/gateway.rb

```diff
@@ -293,6 +293,21 @@ def create_tax_transaction(payment_intent_id, tax_calculation_id)
       end
     end
 
+    def attach_customer_to_credit_card(user)
+      payment_method_id = user&.default_credit_card&.gateway_payment_profile_id
+      return if payment_method_id.blank? || user&.default_credit_card&.gateway_customer_profile_id.present?
+
+      customer = fetch_or_create_customer(user: user)
+      return if customer.blank?
+
+      send_request { Stripe::PaymentMethod.attach(payment_method_id, { customer: customer.profile_id }) }
+
+      user.default_credit_card.update(gateway_customer_profile_id: customer.profile_id, gateway_customer_id: customer.id)
+    rescue Stripe::StripeError => e
+      Rails.error.report(e, context: { user_id: user.id }, source: 'spree_stripe')
```

Comment:
I think we should also add gateway id and/or store id to the context so in multi-store case it'd be easier to identify what Stripe integration has this issue